doc/dev/pylint_checking.md

```diff
@@ -0,0 +1,71 @@
+
+Check each SDK's `Run Pylint Next` output in the [test-weekly CI pipeline](https://dev.azure.com/azure-sdk/internal/_build?pipelineNameFilter=python%20*%20tests-weekly). If there is no corresponding test-weekly pipeline, run `next-pylint` locally with `tox` as described in [How to run Pylint?](#how-to-run-pylint). In order to ensure that the SDK pipeline will not break when pylint is updated, make sure to address all pylint warnings present. 
+
+See below for more information on pylint checkers and how to ignore warnings if needed.
+
+# General Guidance 
+
+### What is pylint?
+
+Pylint is a set of (mostly) [astroid](https://pylint.pycqa.org/projects/astroid/en/latest/index.html) based checkers that run static analysis on your code and check that your formatting aligns with Python's style guide (i.e [PEPs](https://peps.python.org/)). 
+
+In the Azure SDK for Python repository, in addition to the standard pylint library, we added custom checkers that help to align our libraries with the [Azure SDK for Python Guidelines](https://guidelinescollab.github.io/azure-sdk/python_introduction.html).
+
+### How to run Pylint?
+
+One way to run pylint is to run at the package level with tox:
+
+    .../azure-sdk-for-python/sdk/eventgrid/azure-eventgrid>tox -e pylint -c ../../../eng/tox/tox.ini
+
+If you don't want to use tox, you can also install and run pylint on its own:
+
+ - If taking this approach, in order to run with the pylintrc formatting and the custom pylint checkers you must also install the custom checkers and `SET` the pylintrc path. 
+
+- Additional information about custom checkers [here](https://github.com/Azure/azure-sdk-tools/blob/main/tools/pylint-extensions/azure-pylint-guidelines-checker/README.md)
+
+        pip install pylint
+        pip install pylint-guidelines-checker --index-url="https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-python/pypi/simple/"
+
+        .../azure-sdk-for-python>SET PYLINTRC="./pylintrc"
+        .../azure-sdk-for-python>pylint ./sdk/eventgrid/azure-eventgrid
+
+    Note that you may see different errors if running a different version of pylint than the one in CI.
+
+
+# Ignoring Pylint Checkers
+
+There are three ways to ignore pylint checkers:
+- If there is a pylint checker within your SDK that you do not think applies, or that you wish to ignore for that specific scenario (i.e protected-access) you can locally ignore it on the specific line of the warning with the comment `# pylint:disable=protected-access`. If a pylint warning appears multiple times within a file you can ignore all related warnings in that file by putting the disable comment at the top of the file.
+- Another way to disable a pylint checker is by putting in under the `disable` list in the `pylintrc` file. This should only happen if the checker is giving too many false positives, or is not applicable to our general library. Any additions to the pylintrc file should be discussed beforehand. 
```

Comment:
Suggest we don't document this as an option. If we do put something under the disable list, it will probably be discussed internally within our team / added by our team.